packages/hash/frontend/src/blocks/page/Comments/CommentThread.tsx

```diff
@@ -0,0 +1,206 @@
+  const inputRef = useRef<CommentTextFieldRef>(null);
+  const threadRef = useRef<HTMLDivElement>(null);
+  const [threadFocused, setThreadFocused] = useState(false);
+  const [inputFocused, setInputFocused] = useState(false);
+  const [expanded, setExpanded] = useState(false);
+  const [inputValue, setInputValue] = useState<TextToken[]>([]);
+
+  const showInput = threadFocused || !!inputValue.length;
+  const showInputButtons =
+    (threadFocused && inputFocused) || !!inputValue.length;
+
+  const cancelSubmit = () => {
+    inputRef.current?.resetDocument();
+    threadRef.current?.focus();
+  };
+
+  const submitComment = async () => {
+    if (!loading && inputValue?.length) {
+      await createComment(comment.entityId, inputValue).then(() => {
+        inputRef.current?.resetDocument();
+      });
+    }
+  };
+
+  const [collapsedReplies, uncollapsibleReplies] = useMemo(() => {
+    const replies = [...comment.replies];
+    const lastItems = replies.splice(
+      replies.length - UNCOLLAPSIBLE_REPLIES_NUMBER,
+      UNCOLLAPSIBLE_REPLIES_NUMBER,
+    );
+    return [replies, lastItems];
+  }, [comment]);
+
+  const preferredName = useMemo(
+    () =>
+      loadingTypeSystem
+        ? ""
+        : comment.author.properties[
+            extractBaseUri(types.propertyType.preferredName.propertyTypeId)
+          ],
+    [loadingTypeSystem, comment.author.properties],
+  );
+
+  return loadingTypeSystem ? null : (
+    <Box
+      ref={threadRef}
+      tabIndex={0}
+      onFocus={() => setThreadFocused(true)}
+      onBlur={() => setThreadFocused(false)}
+      sx={({ palette, boxShadows }) => ({
+        width: 320,
+        background: palette.white,
+        borderRadius: 1.5,
+        boxShadow: boxShadows.md,
+        marginBottom: 4,
+        outline: "none",
+      })}
+    >
+      <CommentBlock key={comment.entityId} comment={comment} />
+
+      {collapsedReplies.length ? (
+        <>
+          <Button
+            variant="tertiary"
+            onClick={() => setExpanded(!expanded)}
+            size="small"
+            sx={({ palette }) => ({
+              minHeight: 0,
+              height: 40,
+              width: 1,
+              borderRadius: 0,
+              border: "none",
+              borderTop: `1px solid ${palette.gray[20]}`,
+              [`.${buttonClasses.endIcon}`]: {
+                ml: 0.5,
+                color: palette.gray[70],
+                fontSize: 20,
+              },
+            })}
+            endIcon={expanded ? <ExpandLessIcon /> : <ExpandMoreIcon />}
+          >
+            {expanded
+              ? "Show fewer responses"
+              : `Show all ${comment.replies.length} responses`}
+          </Button>
+          <Collapse in={expanded}>
+            {collapsedReplies.map((reply) => (
+              <Box
+                key={reply.entityId}
+                sx={{
+                  borderTop: ({ palette }) => `1px solid ${palette.gray[20]}`,
```

Comment:
Is `CommentBlock` used elsewhere? Can this styling be moved inside of it?

Reply:
It's being used only in `CommentThread`, but the first instance doesn't have a top border.

Reply:
Use CSS to remove it from the first one?

Reply:
Done in 66de3df17535c7548e06f905a5f65f008f7cd4f9